TestBenches/TrackBuilder_test.cpp

```diff
@@ -1,21 +1,82 @@
 // Test bench for TrackBuilder
 #include "TrackBuilderTop.h"
 
-#include <vector>
-#include <algorithm>
-#include <iterator>
-#include <cstring>
-
+#include "Macros.h"
 #include "FileReadUtility.h"
-#include "Constants.h"
 
 const int nevents = 100;  //number of events to run
 
 using namespace std;
 
+// No macros can be defined from the command line in the case of C/RTL
+// cosimulation, so we define defaults here.
+#if !defined SEED_
+  #define SEED_ L1L2_
+#endif
+#if !defined MODULE_
+  #define MODULE_ FT_L1L2_
+#endif
+#if !defined TOP_FUNC_
+  #define TOP_FUNC_ TrackBuilder_L1L2
+#endif
+
+#if SEED_ == L1L2_
+  constexpr unsigned kNBarrelStubs = 4;
+  constexpr unsigned kNDiskStubs = 4;
+#elif SEED_ == L2L3_
+  constexpr unsigned kNBarrelStubs = 3;
+  constexpr unsigned kNDiskStubs = 4;
+#elif SEED_ == L3L4_
+  constexpr unsigned kNBarrelStubs = 4;
+  constexpr unsigned kNDiskStubs = 2;
+#elif SEED_ == L5L6_
+  constexpr unsigned kNBarrelStubs = 4;
+  constexpr unsigned kNDiskStubs = 0;
+#else
+#  error "Undefined seed"
+#endif
+
+typedef TrackFit<kNBarrelStubs, kNDiskStubs> TrackFit_t;
+typedef TrackFitMemory<kNBarrelStubs, kNDiskStubs> TrackFitMemory_t;
+
+template<int I>
+void setBarrelStubs(TrackFit_t &track, const TrackFit_t::BarrelStubWord stubWords[][kMaxProc], const unsigned i) {
+  track.setBarrelStubWord<I>(stubWords[I][i]);
+  setBarrelStubs<I - 1>(track, stubWords, i);
+}
+
+template<>
+void setBarrelStubs<-1>(TrackFit_t &track, const TrackFit_t::BarrelStubWord stubWords[][kMaxProc], const unsigned i) {
+}
+
+template<int I>
+void setDiskStubs(TrackFit_t &track, const TrackFit_t::DiskStubWord stubWords[][kMaxProc], const unsigned i) {
+  track.setDiskStubWord<I + kNBarrelStubs>(stubWords[I][i]);
+  setDiskStubs<I - 1>(track, stubWords, i);
+}
+
+template<>
+void setDiskStubs<-1>(TrackFit_t &track, const TrackFit_t::DiskStubWord stubWords[][kMaxProc], const unsigned i) {
+}
+
+template<int I>
+void compareStubsWithFile(int &err, ifstream &fout, const int pos, const TrackFitMemory_t &tracksMem, const unsigned ievt) {
+  fout.clear(), fout.seekg(pos);
+  stringstream ss("");
+  ss << "\nStub " << (I) << " word";
+  err += compareMemWithFile<TrackFitMemory_t,16,16,TrackFit_t::kTFStubRZResidLSB(I),TrackFit_t::kTFStubValidMSB(I)>(tracksMem, fout, ievt, ss.str(), true);
```

Comment:
Maybe put the "16"s in a constant that has a descriptive name somewhere?  (same goes for L74)

Reply:
Done.